lib/SIL/Utils/OptimizationRemark.cpp

```diff
@@ -167,7 +169,9 @@ static SourceLoc inferOptRemarkSearchBackwards(SILInstruction &i) {
   for (auto &inst : llvm::make_range(std::next(i.getReverseIterator()),
                                      i.getParent()->rend())) {
     auto loc = inst.getLoc();
-    if (!bool(loc))
+    if (auto inlinedLoc = inst.getDebugScope()->getOutermostInlineLocation())
```

Comment:
The instruction already has a valid location but you're using the call site's location instead? I'm confused. Can the rules for inferring debug locations be concisely listed somewhere explaining why we have those rules?

Reply:
Ok, after talking offline. The purpose of this is to always place the remark itself in the outermost inline scope. Future work will preserve the original debug scope so we know which function the retain originated from.

Reply:
>Future work will preserve the original debug scope so we know which function the retain originated from Now I'm confused — isn't that info already present in the form of the scopes pointed to by the inlinedAt: link?

Reply:
When the retain originates from an inlined call, I think we're just dropping that information now.

Reply:
@adrian-prantl in many cases we are just dropping this information in the optimizer when we are performing code motion and other optimizations. We should talk offline about this.